### ISSUE TITLE
how to use loss function for training?

### ISSUE THREAD

**Issue by lishiyu0088** [fulliu/lc#3]
Dear author,

I get little confused reading the loss function.
    def sparse_pose_loss(self, cfg, gt_dict, out_dict):
        pts2d, pts2d_std = itemgetter('pts2d','pts2d_std')(out_dict)
        pts2d_inv_std = 1 / pts2d_std
        pose_best, K, pts3d, bbox_3d = itemgetter('pose_best', 'out_K', 'pts3d', 'bbox_3d')(gt_dict)
        loss_cov = Loss_cov_mixed(K, pose_best, pts3d, pts2d, pts2d_inv_std, None, bbox_3d = bbox_3d)
        return loss_cov.mean()
I saw in the code bbox_3d is necessary to calculate linear covariance loss. Can I use pts3d to replace bbox3d? 
How is pts2d_std calculated?

Thank you very much.

**Comment by fulliu** [comment]
You can use pts3d to replace bbox3d, it should work.
In your case, "pts2d_std" is computed by the DSNT layer ("softargmax_2d_std" in ptnet.py) from the heatmaps.

**Comment by lishiyu0088** [comment]
Thank you very much.
